superset-frontend/packages/superset-ui-chart-controls/src/utils/disabledSectionRules.ts

```diff
@@ -0,0 +1,44 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ensureIsArray, GenericDataType } from '@superset-ui/core';
+
+export enum SectionRuleType {
+  X_AXIS_TEMPORAL = 'XAxisTemporal',
+}
+
+export function isSectionDisabled(rule: SectionRuleType, exploreState: any) {
```

Comment:
It is but we are in chart controls here and I am not seeing a way to import it here other than replicating it

Reply:
Only now I noticed that the logic was inside `superset-ui-chart-controls`. It seems the logic is inverted because an Explore component is calling a function in the generic control here:

```
// src/explore/components/ControlPanelsContainer.ts
const sectionsToExpand = (
  sections: ControlPanelSectionConfig[],
  exploreState: ExplorePageState['explore'],
): string[] =>
  // avoid expanding time section if datasource doesn't include time column
  sections.reduce((acc, section) => {
    const isDisabled = section?.setDisabled?.call(section, exploreState);
    ...
  }, [] as string[]);
```
That's why the control interface ended up with an `exploreState`, even though it shouldn't know anything about Explore. Another way of doing that is moving the validation logic to `ControlPanelsContainer` that is in Explore domain because that's the part that needs to be disabled. If it's disabled, you don't even need to render the controls.

Reply:
@michael-s-molina we have this same pattern for the visibility of controls. I am just replicating the same pattern here for disabling **control sections**. If we want to change this pattern we can do it but I think that's going to be a different discussion. For now, I believe it makes sense to be consistent with the existing design to avoid even more fragmentation.

Reply:
@michael-s-molina we have the same pattern for the visibility of controls. I am just implementing the same pattern for the **control sections** for consistency reasons. The validation logic is encapsulated at the section level because the Explore does not have the context around the section itself in order to decide whether it should or shouldn't be disabled. We would need a map of the sections and their requirements at the Explore level. That's why the validation is currently encapsulated at the section level and that's the same reason it was done that way for the controls I guess.

Reply:
In the future we could create a set of rules at the controls and section levels and implement the validation logics in Explore. In that way the controls and sections only list their requirements but are not responsible for the validation. However, that requires a bigger refactor.